nep-29.mediawiki

```diff
@@ -0,0 +1,60 @@
+<pre>
+  NEP: 29
+  Title: Contract deployment/update callback function
+  Author: Roman Khimov <roman@nspcc.ru>
+  Type: Standard
+  Status: Accepted
+  Created: 2024-07-18
+</pre>
+
+==Abstract==
+
+This proposal standardizes the callback function that can be implemented by contracts to have some code executed right after initial deployment or update.
+
+==Motivation==
+
+Contracts may need to run some initialization/update code right after their deployment or upgrade. Neo N3 has provided a mechanism for that since version 3.0, but it was never properly standardized, so this proposal fixes it.
+
+==Specification==
+
+Neo N3 has a native <code>ContractManagement</code> contract that is used for contract deployment and update via <code>deploy</code> and <code>update</code> methods of it. After settling with the new NEF and manifest both of them can invoke a special <code>_deploy</code> method if it's implemented by the contract.
```

Comment:
Neo N3 has a native <code>ContractManagement</code> contract that is used for contract deployment (<code>deploy</code>) and update (<code>update</code>) methods. After settling with the new NEF and Manifest, both of them can invoke a special <code>_deploy</code> method (if it's implemented by the contract).

Reply:
@roman-khimov, consider this new sentence.

One question.
"both of them can invoke a special¨, you mean, if the developer implements the _deploy method, it is still optional to call it from Deploy/Update,right?
It is not automatically and compulsory, correct?

Reply:
> consider this new sentence.

It was phrased like that in one of the earlier drafts, but I wanted to avoid "deployment (deploy)/update (update)". Pure stylistic, I like the way it's in the PR, but I can change it as well if there is a wider agreement that your version is better.

> if the developer implements the _deploy method

Then it will _always_ be called. Maybe `s/can/will/` makes sense here, a bit stronger causality (if implemented -> will invoke).

Reply:
Fixed. Now we need more approvals.